sofar/sofar.py

```diff
@@ -0,0 +1,2125 @@
+            str
+                Version string, e.g., ``'1.0'``.
+
+        Returns
+        -------
+        convention : dict
+            The SOFA convention as a dictionary
+        """
+        # check input
+        if not isinstance(convention, str):
+            raise TypeError(("Convention must be a string "
+                             f"but is of type {type(convention)}"))
+
+        # get and check path to json file
+        paths = list_conventions(False, "path")
+        path = [path for path in paths
+                if os.path.basename(path).startswith(convention + "_")]
+
+        if not len(path):
+            raise ValueError(
+                (f"Convention '{convention}' not found. See "
+                 "sofar.list_conventions() for available conventions."))
+
+        # select the correct version
+        if version == "latest":
+            path = path[-1]
+        else:
+            versions = [p.split('_')[1][:-5] for p in path]
+            if version not in versions:
+                raise ValueError((
+                    f"Version {version} not found. "
+                    f"Available versions are {versions}"))
+            path = path[versions.index(version)]
+
+        # read convention from json file
+        with open(path, "r") as file:
+            convention = json.load(file)
+
+        # replace ':' and '.' in key names by '_'
+        convention = {
+            key.replace(':', '_'): value for key, value in convention.items()}
+        convention = {
+            key.replace('.', '_'): value for key, value in convention.items()}
+
+        return convention
+
+    def _convention2sofa(self,  mandatory):
+        """
+        Use SOFA convention to create attributes with default values.
+
+        Parameters
+        ----------
+        mandatory : bool
+            Flag to indicate if only mandatory fields are to be included.
+        """
+
+        # populate the SOFA file
+        for key in self._convention.keys():
+
+            # skip optional fields if requested
+            if not self._mandatory(self._convention[key]["flags"]) and mandatory:
+                continue
+
+            # get the default value
+            default = self._convention[key]["default"]
+            if isinstance(default, list):
+                ndim = len(self._convention[key]["dimensions"].split(", ")[0])
+                default = _nd_array(default, ndim)
+
+            # create attribute with default value
+            setattr(self, key, default)
+
+        # write API and date specific fields (some read only)
+        now = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        self._protected = False
+        self.GLOBAL_DateCreated = now
+        self.GLOBAL_DateModified = now
+        self.GLOBAL_APIName = "sofar SOFA API for Python (pyfar.org)"
+        self.GLOBAL_APIVersion = sf.__version__
+        self.GLOBAL_ApplicationName = "Python"
+        self.GLOBAL_ApplicationVersion = platform.python_version()
```

Comment:
```suggestion
        self.GLOBAL_ApplicationVersion = platform.python_version() + f" [{platform.python_implementation()} - {platform.python_compiler()}]"
```
I'd save the python implementation as well as the compiler version as well. This can be quite important when dealing with issues caused by different platforms and compiler versions.

Reply:
good point - changed :)